pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -39,69 +41,81 @@
-    private final Long ddpInstanceId;
+    private Long ddpInstanceId;
 
-    private final String ddpParticipantId;
+    private String ddpParticipantId;
 
     @ColumnName(DBConstants.PARTICIPANT_ID)
-    private final Long participantId;
+    private Long participantId;
 
     @ColumnName(DBConstants.FILE_NAME)
-    private final String fileName;
+    private String fileName;
 
     @ColumnName(DBConstants.MIME_TYPE)
-    private final String mimeType;
+    private String mimeType;
 
-    private final String bucket;
+    private String bucket;
 
-    private final String blobPath;
+    private String blobPath;
 
     @ColumnName(DBConstants.CREATED_BY_USER_ID)
-    private final Long createdByUserId;
+    private Long createdByUserId;
 
     @ColumnName(DBConstants.CREATED_AT)
-    private final long createdAt;
+    private Long createdAt;
 
     @ColumnName(DBConstants.DELETED_BY_USER_ID)
-    private final Long deletedByUserId;
+    private Long deletedByUserId;
 
     @ColumnName(DBConstants.DELETED_AT)
-    private final long deletedAt;
+    private Long deletedAt;
 
     @ColumnName(DBConstants.IS_VIRUS_FREE)
     private Boolean isVirusFree;
 
+    @ColumnName(DBConstants.SENT_AT)
+    private Long sentAt;
+
     private static final String SQL_SELECT_DOCUMENTS_BASE = "SELECT sD.id, p.ddp_instance_id, "
             + "p.ddp_participant_id, p.participant_id, sD.file_name, sD.mime_type, sD.bucket, sD.blob_path, "
-            + "sD.created_by_user_id, sD.created_at, sD.deleted_by_user_id, sD.deleted_at, sD.is_virus_free "
+            + "sD.created_by_user_id, sD.created_at, sD.deleted_by_user_id, sD.deleted_at, sD.is_virus_free, t.created_date  "
             + "FROM  somatic_documents sD "
             + "LEFT JOIN ddp_instance as ddp on (ddp.ddp_instance_id = sD.ddp_instance_id) "
-            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) ";
+            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) "
+            + "LEFT JOIN ddp_survey_trigger t on (t.survey_trigger_id = sD.trigger_id) "
+    ;
 
     private static final String SQL_SELECT_DOCUMENTS_BY_REALM = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE ddp.instance_name = ?";
 
     private static final String SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE ddp.instance_name = ? AND p.ddp_participant_id = ?";
 
+    private static final String SQL_SELECT_DOCUMENT_BY_REALM_PTPT_DOCID = SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT
+            + " AND sD.id = ?";
+
     private static final String SQL_SELECT_DOCUMENT_BY_ID_AND_REALM = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE sD.id = ? AND ddp.instance_name = ?";
 
     private static final String SQL_SELECT_DOCUMENT_BY_BUCKET_AND_PATH = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE sD.bucket = ? AND sD.blob_path = ?";
 
+    private static final String SQL_UPDATE_SENT_AT_TIME = "UPDATE somatic_documents AS sD "
```

Comment:
Added to always update the record with the latest trigger id so we're getting the last time the document was sent.